doc: clarified description of k_uptime_delta

nicbra opened the pull request:
Description added in docs, and updated the header to clarify that the function modifies the argument.

Resolves #100697

peter-mitsis commented:
I like the content of the proposed change. Once the compliance check failure is fixed, I will be happy to give this a +1.

nicbra commented:
> I like the content of the proposed change. Once the compliance check failure is fixed, I will be happy to give this a +1.

It should now pass the compliance check

nicbra committed:
doc: clarified description of k_uptime_delta

Description added in docs.
Header updated to clarify that the function modifies the argument.

Signed-off-by: Nicholas Braaten <git@nicbra.no>
